tasks/docker-engine.yml

```diff
@@ -19,19 +19,60 @@
     repo: 'deb https://apt.dockerproject.org/repo ubuntu-{{ ansible_distribution_release }} main'
     update_cache: yes
 
+- name: see if docker is already installed
+  stat: path=/usr/bin/docker get_checksum=False
+  register: docker_exec_stat_result
+  always_run: True 
 
-# While wildcarding the version is cute, in practice, there only seems to be one
-# version of Docker available at a time. I'm not sure what will happen on the next
-# release. Alternatively, we could install LXC Docker, which does have many more
-# versions, but not the most recent versions.
+- name: Determine installed version
+  command: docker --version
+  changed_when: False
+  always_run: True
+  register: docker_version_result
+  when: "{{ docker_exec_stat_result.stat.exists }}"
 
+# "docker --version" output is formatted like this:
+#
+#     Docker version 1.10.2, build c3959b1
+#
+- name: store installed docker version
+  set_fact:
+    installed_docker_version: "{{ docker_version_result.stdout
+                                  | regex_replace('^Docker version ', '')
+                                  | regex_replace(', build [0-9a-f]+$', '') }}"
+  when: "{{ docker_exec_stat_result.stat.exists }}"
```

Comment:
Not strictly - but it's a read-only operation, it's almost instant, and having this info in the log might be useful for troubleshooting.